### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jperon**

### CODE REVIEW

**Comment by uliska** [comment]
@jperon #156 should be fixed before merging this PR.

File: test.tex [hunk 1]
```diff
@@ -19,13 +19,14 @@
 
 % It's possible to wrap `ly` within another environment, using \ly[] and \endly.
```
Comment (rpspringuel) [inline test.tex]: Comment no longer matches content.

**Comment by jperon** [comment]
e56a803 didn't work when there where several options with braces, for example:

```TeX
\lilypondfile[includepaths={./,test}, print-only={3, 1, 5}]{ly/Ave Maria De Lourdes/Ave Maria De Lourdes}
```

So I pushed another fix.

**Comment by jperon** [comment]
@uliska Could you please test it ? If it works, I think we can merge.

**Comment by jperon** [comment]
In fact, I've just seen that #139 isn't really fixed; so please don't merge at once.

**Comment by uliska** [comment]
#156 can be considered fixed, but I won't merge as per your last comment.

**Comment by jperon** [comment]
Now that should be good. #139 isn't perfectly fixed, but I think it won't be totally fixable as is. The reason is that LuaTeX's `process_input_buffer` needs a newline to begin processing lines; but LaTeX deletes the newline after `\begin{env}` when `env` has only optional arguments and no `[]` is specified.

**Pull request merged**